executor/src/witgen/block_processor.rs

```diff
@@ -185,7 +185,7 @@ mod tests {
     }
 
     fn solve_and_assert<T: FieldElement>(src: &str, asserted_values: &[(usize, &str, u64)]) {
-        let query_callback = |_: &str| -> Option<T> { None };
+        let query_callback = |_: &str| -> Result<Option<T>, String> { Ok(None) };
```

Comment:
In `mock_prover.rs` we panic instead of returning `None`. I think that's better!

How about adding this function:
```rust
pub fn empty_query_callback<T>() -> fn(&str) -> Result<Option<T>, String> {
    |_: &str| -> Result<Option<T>, String> { unreachable!() }
}
```

Then we can call it in all places where we need something like this.

Reply:
Added this as `unused_query_callback` and squashed.